File: app/routes/api/v1/internal/create_announcement.py
from flask import request
from . import internal
from .....static.python.mongodb.create import createAnnouncement


@internal.route("/create-announcement", methods=["POST"])
def create_announcement():
    """
    Create an announcement.
    """
    data = request.get_json()
    print(data)
    createAnnouncement(data)
    return "success", 200

File: app/static/js/createObjects/createAnnouncement.js
function createAnnouncement() {
    const announcement = {
        title: document.getElementById("announcementname").value,
        content: document.getElementById("message").value,
        author: document.getElementById("author").value,
        author_pic: document.querySelector("#navbar trigger logo").getAttribute("image"),
        course: document.getElementById("course_id").innerText,
    };

    $.ajax({
        type: "POST",
        url: "/api/v1/internal/create-announcement",
        data: JSON.stringify(announcement),
        dataType: "json",
        contentType: "application/json",
        done: function (data) {
            alert('Successfully created announcement');
        },
        error: function (error) {
            alert(error);
        }
    });
}